File: articles/decision-records-and-accountable-judgment/r/run_all_decision_record_workflows.R
# run_all_decision_record_workflows.R
# Run all R workflows for Decision Records and Accountable Judgment.

args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)

if (length(file_arg) > 0) {
  script_path <- normalizePath(sub("^--file=", "", file_arg[1]), mustWork = TRUE)
  article_root <- normalizePath(file.path(dirname(script_path), ".."), mustWork = TRUE)
} else {
  article_root <- getwd()
}

r_dir <- file.path(article_root, "r")

scripts <- c(
  "decision_records_accountable_judgment_workflow.R",
  "record_completeness_profiles.R",
  "evidence_traceability_report.R",
  "assumption_monitoring_gaps.R",
  "accountability_review_priority.R",
  "review_trigger_tables.R"
)

for (script in scripts) {
  path <- file.path(r_dir, script)
  message("Running ", path)
  source(path)
}

message("All R decision-record workflows completed.")

File: articles/decision-records-and-accountable-judgment/python/run_all_decision_record_workflows.py
#!/usr/bin/env python3
"""Run all Python workflows for Decision Records and Accountable Judgment."""

from __future__ import annotations

from pathlib import Path
import subprocess
import sys

ARTICLE_ROOT = Path(__file__).resolve().parents[1]
PYTHON_DIR = ARTICLE_ROOT / "python"

SCRIPTS = [
    "decision_record_completeness_score.py",
    "evidence_traceability_checker.py",
    "assumption_criticality_audit.py",
    "dissent_preservation_audit.py",
    "review_trigger_monitor.py",
    "post_decision_learning_loop.py",
    "decision_record_exporter.py",
    "decision_records_accountable_judgment_audit.py",
]


def main() -> None:
    for script in SCRIPTS:
        path = PYTHON_DIR / script
        print(f"Running {path.name}...")
        subprocess.run([sys.executable, str(path)], check=True)
    print("All Python decision-record workflows completed.")


if __name__ == "__main__":
    main()
